tests/deletion_pool_test.go

```diff
@@ -358,3 +362,151 @@ func TestNoopReceivesPendingDeletionUpdates(t *testing.T) {
 		c.Sx(`A002 OK`)
 	})
 }
+
+type eventWaiter struct {
```

Comment:
given we created a db_utils.go to hold just one function, i think this eventWaiter and its methods deserve their own file.

Reply:
Fair point. Fixed.